clients/feeder/timeouts_test.go

```diff
@@ -0,0 +1,105 @@
+package feeder
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseTimeouts(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   []string
+		want    []time.Duration
+		wantErr bool
+	}{
+		{
+			name:  "empty input",
+			input: []string{},
+			want:  []time.Duration{defaultTimeout},
+		},
+		{
+			name:  "single value",
+			input: []string{"5s"},
+			want:  []time.Duration{5 * time.Second},
+		},
+		{
+			name:  "multiple values",
+			input: []string{"5s", "7s", "10s"},
+			want:  []time.Duration{5 * time.Second, 7 * time.Second, 10 * time.Second},
+		},
+		{
+			name:    "invalid duration",
+			input:   []string{"5s", "invalid", "10s"},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := ParseTimeouts(tt.input)
+			if tt.wantErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestGetTimeouts(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      []time.Duration
+		maxRetries int
+		want       Timeouts
+	}{
+		{
+			name:       "empty input",
+			input:      []time.Duration{},
+			maxRetries: 4,
+			want: Timeouts{
+				curTimeout: 0,
+				timeouts:   []time.Duration{5 * time.Second, 8 * time.Second, 12 * time.Second, 18 * time.Second, 27 * time.Second},
+			},
```

Comment:
From this test it seems that the timeout array is still connected with the amount of retries. Max retires means the max amount of continuous retries, it should have 0 influence on the timeouts array.

The timeout array should be of fixed size 30, max retries can be 5 or a 1000, it shouldn't affect it. 

If you have 5 max retries and you are step `i`:
1. You fail 4 times, mean you are now on `i + 4`:
2. You have one succesful trie, you are now on `i + 3`
3. You fail 4 times, mean you are now on `i + 7`
4. and so on

Even if the max retries it's 5 it still possible to use more than that.

Reply:
Yess, I understand this. The maxRetries in this test is just an arbitrarily small number, I can manually pass for example 5, so its more readable and clear (that timeouts list is in no way connected to the maxRetries number).

Reply:
Or just test it against 30 values, I just realized, that after the refactorings the `count` param is used only for tests

Reply:
I think is not necessary to write the 30 values, if it is not related then all good, I think ti shouldn't be related in the tests either.